app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -14,16 +14,31 @@ def google_oauth2
     end
   end
 
+  def facebook
+    if user.present?
+      user.accept_invitation! if invitation_token
+      flash[:notice] = I18n.t('devise.omniauth_callbacks.success',
+                              kind: 'Facebook')
+      sign_in_and_redirect @user, event: :authentication
+    else
+      redirect_to new_user_session_path, notice: t('omniauth.access_denied')
+    end
+  end
+
   private
 
   def user
-    @user ||= User.find_for_google_oauth2(request.env['omniauth.auth'])
+    @user ||= User.find_for_oauth(request.env['omniauth.auth'])
   end
 
   def google_avatar
     request.env['omniauth.auth']&.[]('info')&.[]('image')
   end
 
+  def facebook_avatar
```

Comment:
It's looking good. a minor refactor here. Could we combine `google_avatar` and `facebook_avatar` into one `omniauth_avatar` method?